Student/Student/Courses/Syllabus/SyllabusViewController.swift

```diff
@@ -0,0 +1,187 @@
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class SyllabusViewController: UIViewController {
+
+    @IBOutlet weak var menuBorder: UIView!
+    @IBOutlet weak var menuBorderHeight: NSLayoutConstraint!
+    @IBOutlet weak var menu: HorizontalMenuView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    var presenter: SyllabusPresenter!
+    var titleView: TitleSubtitleView!
+    var courseID: String = ""
+    var color: UIColor?
+    var syllabus: CoreWebView?
+    var assignments: AssignmentListViewController?
+
+    enum MenuItem: Int {
+        case syllabus, assignments
+    }
+
+    static func create(courseID: String) -> SyllabusViewController {
+        let vc = loadFromStoryboard()
+        vc.courseID = courseID
+        return vc
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        presenter = SyllabusPresenter(courseID: courseID, view: self)
+        view.backgroundColor = UIColor.white.ensureContrast(against: .named(.white))
+
+        configureTitleView()
+        configureSyllabus()
+        configureAssignments()
+        configureMenu()
+
+        scrollView.delegate = self
+        presenter.viewIsReady()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        scrollView.contentSize = CGSize(width: view.bounds.size.width * 2, height: scrollView.bounds.size.height)
+    }
+
+    // MARK: - Setup
+
+    func configureMenu() {
+        menu.delegate = self
+        menuBorderHeight.constant = 1.0 / UIScreen.main.scale
+    }
+
+    func configureTitleView() {
+        titleView = TitleSubtitleView.create()
+        navigationItem.titleView = titleView
+        titleView.title = NSLocalizedString("Course Syllabus", comment: "")
+    }
+
+    func configureSyllabus() {
+        syllabus = CoreWebView(frame: CGRect.zero)
+        syllabus?.accessibilityLabel = "SyllabusPage.syllabusWebView"
+        guard let syllabus = syllabus else { return }
+        scrollView.addSubview(syllabus)
+        syllabus.addConstraintsWithVFL("H:|[view(==superview)]")
+        syllabus.addConstraintsWithVFL("V:|[view(==superview)]|")
+    }
+
+    func configureAssignments() {
+        assignments = AssignmentListViewController(courseID: courseID, sort: GetAssignments.Sort.dueAt)
+        guard let assignments = assignments else { return }
+        embed(assignments, in: scrollView) { [weak self] (child, _) in
+            guard let syllabus = self?.syllabus else { return }
+            child.view.accessibilityLabel = "SyllabusPage.assignmentList"
+            child.view.addConstraintsWithVFL("H:[syllabus][view(==superview)]|", views: ["syllabus": syllabus])
+            child.view.pinToTopAndBottomOfSuperview()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
```

Comment:
Nice job testing and handling rotation.

Did you consider using UIPageViewController for this screen? I wonder if it handles rotation out of the box..

Reply:
I did not.  I ended up getting this working pretty reliably.  Also, the PagedViewController does not work well with just views (the webView is not a ViewController, it's just the `CoreWebView`).  Would you know how it handles rotation?

Reply:
Ah, makes sense. I don't know how PageViewController handles rotation.